language/diem-tools/writeset-transaction-generator/src/main.rs

```diff
@@ -42,11 +47,64 @@ enum Command {
     /// Create a release writeset by comparing local Diem Framework against a remote blockchain state.
     #[structopt(name = "create-release")]
     CreateDiemFrameworkRelease {
+        /// ChainID to distinguish the diem network. e.g: PREMAINNET
+        chain_id: ChainId,
         /// Public JSON-rpc endpoint URL.
+        // TODO: Get rid of this URL argument once we have a stable mapping from ChainId to its url.
         url: String,
         /// Blockchain height
         version: u64,
     },
+    /// Verify if a blob is generated by the checked-in artifact.
+    #[structopt(name = "verify-release")]
+    VerifyDiemFrameworkRelease {
+        /// ChainID to distinguish the diem network. e.g: PREMAINNET
+        chain_id: ChainId,
+        /// Public JSON-rpc endpoint URL.
+        url: String,
+        /// Path to the serialized bytes of WriteSet.
+        #[structopt(parse(from_os_str))]
+        writeset_path: PathBuf,
+    },
+}
+
+/// Checkin the release artifact to a local file to make sure the release process is reproducible
+#[derive(Serialize, Deserialize)]
+struct ReleaseArtifact {
+    chain_id: ChainId,
+    // TODO: Deprecate after release
```

Comment:
Do we need to keep this?

Reply:
I thought keeping this item will be desirable because we can prevent the manual error from inputting a chain_id with a different url.